### ISSUE TITLE
Load model Error in Heterogenous Graphs

### ISSUE THREAD

**Issue by shubham-gupta-iitr** [pyg-team/pytorch_geometric#3640]
## 🐛 Bug

While using the to_hetero method to create heterogenous models, I am not able to load the model if I use torch.save and torch.load. Is there something I am missing here?
### To Reproduce

````python
import torch
from torch_geometric.nn import TopKPooling
from torch_geometric.nn import global_mean_pool as gap, global_max_pool as gmp
import torch.nn.functional as F
from torch_geometric.nn import SAGEConv
from torch_geometric.nn.norm import BatchNorm
from torch_geometric.nn.dense import Linear

class ToyNet(torch.nn.Module):
    def __init__(self):
        super(ToyNet, self).__init__()
        self.conv1 = SAGEConv((-1, -1), 2048)
        self.conv2 = SAGEConv((-1, -1), 1024)
        self.conv3 = SAGEConv((-1, -1), 512)
        self.conv4 = SAGEConv((-1, -1), 256)
        self.conv5 = SAGEConv((-1, -1), 256)
        self.conv6 = SAGEConv((-1, -1), 128)
        self.lin1 = torch.nn.Linear(128, 128)
        self.lin2 = torch.nn.Linear(128, 64)
        self.lin3 = torch.nn.Linear(64, 2)

        self.blinq = torch.nn.BatchNorm1d(512)
        self.blinc = torch.nn.BatchNorm1d(512)
        self.blini = torch.nn.BatchNorm1d(512)

        self.act1 = torch.nn.ReLU()
        self.act2 = torch.nn.ReLU()  
        self._softmax = torch.nn.Softmax(dim=-1)      
        print("Just want to be sure")

    def forward(self, x, edge_index):
        x = F.relu(self.conv1(x, edge_index))
        
        x = F.relu(self.conv2(x, edge_index))
        x = F.relu(self.conv3(x, edge_index))
        x = F.relu(self.conv4(x, edge_index))
        x = F.relu(self.conv5(x, edge_index))
        x = F.relu(self.conv6(x, edge_index))
        x = self.lin1(x)
        x = self.act1(x)
        x = self.lin2(x)
        x = self.act2(x)      
        x = F.dropout(x, p=0.5, training=self.training)
        x_log = self.lin3(x)
        x = self._softmax(x_log)
        return x_log

toy_model = ToyNet()
graph_meta = (['txt_src', 'img_src', 'ques'], [('ques','link1','txt_src'), ('ques','link2','img_src'), 
('txt_src','link3','ques'), ('img_src','link4','ques')])

toy_model = to_hetero(toy_model, graph_meta)
# Run some epochs
torch.save({'epoch': epoch,
            'state_dict': toy_model.state_dict()},
            'train.ckpt')

# During inference
toy_model.load_state_dict(torch.load('train.ckpt'))
```

### Expected behavior

Expect the model to load the saved model.

### Environment

* PyG version (`torch_geometric.__version__`): 2.0.2
* PyTorch version: (`torch.__version__`): 1.9.1
* OS (*e.g.*, Linux): Linux
* Python version (*e.g.*, `3.9`): 3.8.5
* CUDA/cuDNN version: 11.1
* How you installed PyTorch and PyG (`conda`, `pip`, source): pip
* Any other relevant information (*e.g.*, version of `torch-scatter`): 

### Additional context

<!-- Add any other context about the problem here -->

**Comment by saiden89** [comment]
Would you please fill the form and post a reproducible example?

**Comment by rusty1s** [comment]
The error for me is
```
ValueError: Attempted to use an uninitialized parameter in <method 'detach' of 'torch._C._TensorBase' objects>. This error happens when you are using a `LazyModule` or explicitly manipulating `torch.nn.parameter.UninitializedParameter` objects. When using LazyModules Call `forward` with a dummy batch to initialize the parameters before calling torch functions
```
Can you confirm? I think there is an issue with saving and loading a `Linear` layer that has uninitialized parameters. I will look into that.

In the meantime, you can either:
* Run your forward loop once to initalize parameters and save the `state_dict` of the model afterwards
* Save the model via `torch.save(model)`

**Comment by shubham-gupta-iitr** [comment]
So, I already save it after running the epochs.

I think what is happening is when we load, there are still some uninitialized parameters because we have not yet defined the graph input sizes. Is there a way to explicitly do it before calling the forward during the inference?

**Comment by rusty1s** [comment]
There shouldn't exist any uninitialized parameter after running for a few epochs. Can you clarify?

**Comment by shubham-gupta-iitr** [comment]
I mean saving works fine. Loading has issues.

**Comment by rusty1s** [comment]
Ah got it. Yeah, you will need to run forward once before loading the model. I'm sorry for that. Will work on a fix ASAP.

**Comment by shubham-gupta-iitr** [comment]
No worries, thanks a lot.

**Comment by rusty1s** [comment]
I pushed a fix via #3651 to master. This should resolve your issues. Let me know if it works for you :)
